third_party/wabt/src/interp/jit/jit-backend.cc

```diff
@@ -186,6 +187,35 @@ static void operandToArg(Operand* operand, JITArg& arg) {
   }
 }
 
+static void floatOperandToArg(sljit_compiler* compiler,
+                              Operand* operand,
+                              JITArg& arg,
+                              unsigned int id) {
+  if (operand->location.type == Operand::Immediate) {
+    Instruction* instr = operand->item->asInstruction();
+
+#if (defined SLJIT_32BIT_ARCHITECTURE && SLJIT_32BIT_ARCHITECTURE)
+    assert((operand->location.value_info & LocationInfo::kSizeMask) == 1);
+    arg.argw = static_cast<sljit_s32>(instr->value().value32);
```

Comment:
Float64 is also possible in 32 bit systems, this assert seems wrong. Have you compiled it on 32 bit? You need to put both halves into the memory and load it or just use the new `sljit_emit_fcopy` function.